### PR TITLE
[CINN] Enable check_cinn on some tests

### PR DESCRIPTION
**Pull request by FisherWY**
### PR types
Others

### PR changes
Others

### Description
card-70979

为以下算子的单测开启With CINN测试：

| 算子名 | 备注 |
| ------ | ----- |
| bitwise | 开启后遇到的问题已修复，[链接](https://github.com/PaddlePaddle/CINN/pull/1404) |
| compare | - |
| shape | - |
| assign_value | - |
| sum | - |
| expand_v2 | - |
| lookup_table | - |
| lookup_table_v2 | - |
| matmul_v2 | CINN支持fp64类型，[链接](https://github.com/PaddlePaddle/CINN/pull/1421) |
| matmul | CINN支持fp64类型，[链接](https://github.com/PaddlePaddle/CINN/pull/1421) |
| norm | Norm反向fp16类型转换，[链接](https://github.com/PaddlePaddle/CINN/pull/1446) |
| one_hot_v2 | - |

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_assign_value_op.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def init_data(self):
         self.attrs["fp32_values"] = [float(v) for v in self.value.flat]
 
     def test_forward(self):
-        self.check_output()
+        self.check_output(check_cinn=True)
```
Comment (zzk0) [inline python/paddle/fluid/tests/unittests/test_assign_value_op.py:52]: check_cinn=self.check_cinn

File: python/paddle/fluid/tests/unittests/test_sum_op.py [hunk 2]
```diff
@@ -351,11 +351,11 @@ def init_kernel_type(self):
 
     def test_check_output(self):
         # new dynamic graph mode does not support unit16 type
-        self.check_output(check_dygraph=False)
+        self.check_output(check_dygraph=False, check_cinn=True)
```
Comment (thisjiang) [inline python/paddle/fluid/tests/unittests/test_sum_op.py]: BF16的不用测试

File: python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py [hunk 3]
```diff
@@ -98,14 +99,15 @@ def setUp(self):
...
-        self.check_output()
+        self.check_output(check_cinn=True)
 
 
 class TestOneHotOp_default_dtype_attr(OpTest):
     def setUp(self):
         self.op_type = 'one_hot_v2'
         self.python_api = one_hot_wrapper
         depth = 10
+        depth_np = np.array(10).astype('int32')
```
Comment (thisjiang) [inline python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py]:
```suggestion
        depth_np = np.array(depth).astype('int32')
```

File: python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py [hunk 4]
```diff
@@ -118,12 +120,12 @@ def setUp(self):
         for i in range(np.product(x.shape)):
             out[i, 0, x[i]] = 1.0
 
-        self.inputs = {'X': (x, x_lod)}
+        self.inputs = {'X': (x, x_lod), 'depth_tensor': depth_np}
```
Comment (thisjiang) [inline python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py]: 为啥要额外加上`depth_np`属性？这破坏了原来的单测啊

File: python/paddle/fluid/tests/unittests/test_norm_op.py [hunk 5]
```diff
@@ -120,11 +120,17 @@ def init_dtype(self):
...
+        )
 
     def test_check_grad(self):
         self.check_grad_with_place(
-            fluid.core.CUDAPlace(0), ['X'], 'Out', max_relative_error=0.05
+            fluid.core.CUDAPlace(0),
+            ['X'],
+            'Out',
+            max_relative_error=0.05,
+            check_cinn=False,
```
Comment (thisjiang) [inline python/paddle/fluid/tests/unittests/test_norm_op.py]: False的就不用特意加了

File: python/paddle/fluid/tests/unittests/test_expand_v2_op.py [hunk 6]
```diff
@@ -245,11 +247,13 @@ def setUp(self):
...
     def test_check_output(self):
         place = core.CUDAPlace(0)
-        self.check_output_with_place(place)
+        self.check_output_with_place(place, check_cinn=True)
 
     def test_check_grad(self):
         place = core.CUDAPlace(0)
-        self.check_grad_with_place(place, ['X'], 'Out', check_prim=True)
+        self.check_grad_with_place(
+            place, ['X'], 'Out', check_prim=True, check_cinn=True
```
Comment (thisjiang) [inline python/paddle/fluid/tests/unittests/test_expand_v2_op.py]: check_prim和check_cinn重复了，有check_prim的就不用加check_cinn了

File: python/paddle/fluid/tests/unittests/test_norm_op.py (hunk 5 above)
Comment (FisherWY) [reply python/paddle/fluid/tests/unittests/test_norm_op.py]: 暂时设置为False是因为cinn侧的梯度计算结果没有做类型转换，现已解决：https://github.com/PaddlePaddle/CINN/pull/1446

File: python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py (hunk 4 above)
Comment (FisherWY) [reply python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py]: CINN不支持该单测的输出类型（将`Out`与`x_lod`包装为一个`tuple`），已跳过该单测

File: python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py (hunk 3 above)
Comment (FisherWY) [reply python/paddle/fluid/tests/unittests/test_one_hot_v2_op.py]: Done

File: python/paddle/fluid/tests/unittests/test_sum_op.py (hunk 2 above)
Comment (FisherWY) [reply python/paddle/fluid/tests/unittests/test_sum_op.py]: Done

**Review by thisjiang — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]
LGTM

**Pull request merged**